### ISSUE TITLE
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.

### ISSUE THREAD

**Issue by andrewzheng1618** [QwenLM/Qwen2#897]
Copy https://qwen.readthedocs.io/zh-cn/latest/getting_started/quickstart.html directly and change 7B to 1.5B only.

from transformers import AutoModelForCausalLM, AutoTokenizer

device = "cuda" # the device to load the model onto

model = AutoModelForCausalLM.from_pretrained(
    "Qwen/Qwen2-1.5B-Instruct",
    torch_dtype="auto",
    device_map="auto",
)
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2-1.5B-Instruct")

prompt = "Give me a short introduction to large language model."
messages = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": prompt},
]
text = tokenizer.apply_chat_template(
    messages,
    tokenize=False,
    add_generation_prompt=True,
)
model_inputs = tokenizer([text], return_tensors="pt").to(device)

generated_ids = model.generate(
    model_inputs.input_ids,
    max_new_tokens=512,
)
generated_ids = [
    output_ids[len(input_ids):] for input_ids, output_ids in zip(model_inputs.input_ids, generated_ids)
]

response = tokenizer.batch_decode(generated_ids, skip_special_tokens=True)[0]

**Comment by jklj077** [comment]
Hi,

The warning can be safely ignored for single sequence inference as in that Quickstart.
You can use `model.generate(**model_inputs, max_new_tokens=512)` instead if there are multiple sequences.
It is shown at the GitHub README or at <https://qwen.readthedocs.io/zh-cn/latest/inference/chat.html>.
